network/p2p/middleware/middleware.go

```diff
@@ -348,14 +339,15 @@ func (m *Middleware) topologyPeers() peer.IDSlice {
 	return peerIDs
 }
 
-// OnDisallowListNotification is called when a new disallow list update notification is distributed.
-// It disconnects from all peers in the disallow list.
-func (m *Middleware) OnDisallowListNotification(notification *p2p.DisallowListUpdateNotification) {
-	for _, pid := range m.peerIDs(notification.DisallowList) {
-		err := m.libP2PNode.RemovePeer(pid)
-		if err != nil {
-			m.log.Error().Err(err).Str("peer_id", pid.String()).Msg("failed to disconnect from blocklisted peer")
-		}
+func (m *Middleware) OnDisallowListNotification(notification *network.DisallowListingUpdate) {
```

Comment:
In this PR, we observed that distributing the disallow-listing notification is unnecessary since it has only one consumer, the `middleware`. Therefore, this PR directly connects the consumer to the notification producer. Given that there's just one consumer, having a distributor broker seems excessive. Additionally, as we’re phasing out the `middleware` in favor of integrating it into the `LibP2PNode` and `network` components. Eventually, only the `network` layer will consume these notifications. This aligns well architecturally with the network layer handling notifications regarding node disallow-listing at the network level.